addons/survey/models/survey.py

```diff
@@ -0,0 +1,1159 @@
+        string='Questions', copy=True)
+    sequence = fields.Integer(string='Page number', default=10)
+    description = fields.Html('Description',
+        help="An introductory text to your page", translate=True,
+        oldname="note")
+
+    # Public methods #
+
+    def copy_data(self, cr, uid, id, default=None, context=None):
+        current_rec = self.read(cr, uid, id, fields=['title'], context=context)
+        title = _("%s (copy)") % (current_rec.get('title'))
+        default = dict(default or {}, title=title)
+        return super(SurveyPage, self).copy_data(cr, uid, id, default,
+            context=context)
+
+
+class SurveyQuestion(models.Model):
+    ''' Questions that will be asked in a survey.
+
+    Each question can have one of more suggested answers (eg. in case of
+    dropdown choices, multi-answer checkboxes, radio buttons...).'''
+    _name = 'survey.question'
+    _description = 'Survey Question'
+    _rec_name = 'question'
+    _order = 'sequence,id'
+
+    # Model fields #
+
+    # Question metadata
+    page_id = fields.Many2one('survey.page', string='Survey page',
+        ondelete='cascade', required=True, default=lambda self: self.env.context.get('page_id'))
+    survey_id = fields.Many2one('survey.survey', related='page_id.survey_id', string='Survey')
+    sequence = fields.Integer(default=10)
+
+    # Question
+    question = fields.Char(string='Question Name', required=True, translate=True)
+    description = fields.Html(help="Use this field to add \
+        additional explanations about your question", translate=True,
+        oldname='descriptive_text')
+
+    # Answer
+    question_type = fields.Selection([('free_text', 'Multiple Lines Text Box'),
+            ('textbox', 'Single Line Text Box'),
+            ('numerical_box', 'Numerical Value'),
+            ('datetime', 'Date and Time'),
+            ('simple_choice', 'Multiple choice: only one answer'),
+            ('multiple_choice', 'Multiple choice: multiple answers allowed'),
+            ('matrix', 'Matrix')], string='Type of Question', size=15, required=True, default='free_text')
+    matrix_subtype = fields.Selection([('simple', 'One choice per row'),
+        ('multiple', 'Multiple choices per row')], string='Matrix Type', default='simple')
+    labels_ids = fields.One2many('survey.label',
+        'question_id', string='Types of answers', oldname='answer_choice_ids', copy=True)
+    labels_ids_2 = fields.One2many('survey.label',
+        'question_id_2', string='Rows of the Matrix', copy=True)
+    # labels are used for proposed choices
+    # if question.question_type == simple choice | multiple choice
+    #                    -> only labels_ids is used
+    # if question.question_type == matrix
+    #                    -> labels_ids are the columns of the matrix
+    #                    -> labels_ids_2 are the rows of the matrix
+
+    # Display options
+    column_nb = fields.Selection([('12', '1'),
```

Comment:
totally unuseful as this value will only be concateneted with strings like "col-md"